src/cascadia/WindowsTerminal/WindowUiaProvider.hpp

```diff
@@ -0,0 +1,47 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- windowUiaProvider.hpp
+
+Abstract:
+- This module provides UI Automation access to the console window to
+  support both automation tests and accessibility (screen reading)
+  applications.
+- Based on examples, sample code, and guidance from
+  https://msdn.microsoft.com/en-us/library/windows/desktop/ee671596(v=vs.85).aspx
+
+Author(s):
+- Michael Niksa (MiNiksa)     2017
+- Austin Diviness (AustDi)    2017
+--*/
+
+#pragma once
+
+#include "../types/WindowUiaProviderBase.hpp"
+
+class WindowUiaProvider final :

```

Comment:
Maybe rename this something specific to the terminal?

Reply:
Eh. Marking as _wont fix_ for now. It does deal with the caption buttons and the top-level uia tree but I want to see what it looks like now that we've messed with some things in TermApp.